### DISCUSSION TITLE
Logitech PRO RACING WHEEL

### DISCUSSION THREAD

**Discussion by Helge102** [berarma/new-lg4ff#69] [General]
I'm thinking about buying a Logitech PRO RACING WHEEL.
Will it work with new-lg4ff?
I don't think anyone can say that due to the lack of a test device, but maybe there was already a successful test.

**Comment by berarma** [comment]
It won't work with new-lg4ff. If it will work in some other way or how hard would it be to make it work, I don't know.

**Comment by Helge102** [comment]
I assumed that.
The device is still very new on the market. I'll just wait and see then.

**Comment by isopix** [comment]
It's new, but Logitech were generally linux friendly in the past.

It's Logitech, so it will be popular for sure and has great
compatiblity with games (like all their previous wheels)

Unlike their previous wheels, this one is really awesome, with no
potentiometers, but hall sensors and load cell. And potentiometers
were the only thing failing in G29 and newer wheels, so this one is
probably also designed to last (or we can hope so).

So it's definitely different device, probably using different
communication protocols.

And this one has built-in USB-hub (3* usb-A ports) for other
accesories (like pedals, shifter, handbrake, etc), unlike some
proprietary connectors (ans some bullshit 'hubs', like Fanatec).

@berarma:
BTW, why new-lg4ff is not upstreamed, if it's mostly ready? You want
to merge it with standard logitech-hid firstst (they are colliding
now, right, and can not be installed at the same time?) or upstream
doesn't like your code, because floating points operations or
something?

**Reply by berarma** [reply]
I foresee a negative response. It's an important change that adds some complexity to a previously very simple driver. And the authors of the current driver haven't expressed good feelings about it.

**Reply by sixtysecondstosmash** [reply]
We're closing in on 3 years since this conversation was last updated. Has there been any progress in getting the G Pro DD added into this driver? I was in the bazzite thread where they merged a lot of wheel drivers into their build, and they advised to reach out to the Logitech driver maintainers so they can pull it in.

It looks like someone has successfully gotten the wheel working on Linux using a different driver called hid-logitech-hidpp. The thread is here https://www.reddit.com/r/linux_gaming/comments/1m3tonb/guide_how_i_got_the_logitech_g_pro_racing_wheel/ , and as both Silverblue and Bazzite are atomic distros, I'm unfamiliar with how to layer it in.

Is that other driver in any way compatible to add into this one?

**Reply by berarma** [reply]
> We're closing in on 3 years since this conversation was last updated. Has there been any progress in getting the G Pro DD added into this driver? I was in the bazzite thread where they merged a lot of wheel drivers into their build, and they advised to reach out to the Logitech driver maintainers so they can pull it in.
> 
> It looks like someone has successfully gotten the wheel working on Linux using a different driver called hid-logitech-hidpp. The thread is here https://www.reddit.com/r/linux_gaming/comments/1m3tonb/guide_how_i_got_the_logitech_g_pro_racing_wheel/ , and as both Silverblue and Bazzite are atomic distros, I'm unfamiliar with how to layer it in.
> 
> Is that other driver in any way compatible to add into this one?

That wheel won't ever work on this driver since it uses a completely different protocol already implemented in hid-logitech-hidpp. It's the same one used by the G920 and G923 XBOX. I'm not sure how good is the compatibility with the G Pro DD though.

For installing the patched driver, I would ask in the help forums of your distribution.

**Reply by PDM1978** [reply]
Gracias al autor y a la comunidad por el trabajo que hacen, estoy con un g923 funcionando bien en Cachyos con el complemento new-lg4ff estaba pensando en comprar el nuevo logitech g rs50 dd supongo que será lo mismo que vienen conversando de que no funcionará con new-lg4ff verdad?

**Reply by sixtysecondstosmash** [reply]
> That wheel won't ever work on this driver since it uses a completely different protocol already implemented in hid-logitech-hidpp. It's the same one used by the G920 and G923 XBOX. I'm not sure how good is the compatibility with the G Pro DD though.

I wasn't necessarily looking to see if this driver in its current state could be used for the DD wheel. I was wondering if it were possible to merge the bits of hid-logitech-hidpp for the wheel base into this one, whatever that would entail (I can barely understand what some of this means). As this driver (new-lg4ff) has already been integrated into Bazzite, Logi DD support would then get it (currently I don't think it's possible any other way since atomic distros can't use DKMS – it's basically Fedora Silverblue/Kinoite).
